azure-pipelines.yml

```diff
@@ -61,27 +31,27 @@ jobs:
     vmImage: ubuntu-18.04
 
   steps:    
-  - template: azure-pipelines-steps-node.yml
-    parameters:
-      nodeVersion: $(nodeVersion)
-  - template: azure-pipelines-steps-test-build.yml
+    - template: azure-pipelines-steps-node.yml
+      parameters:
+        nodeVersion: $(nodeVersion)
+
+    - template: azure-pipelines-steps-test-build.yml
 
-  - task: PublishPipelineArtifact@1
-    inputs:
-      targetPath: 'node/_build'
-      artifactType: 'pipeline'
-      artifactName: 'npm-package'
+    - task: PublishPipelineArtifact@1
+      inputs:
+        targetPath: 'node/_build'
+        artifactType: 'pipeline'
+        artifactName: 'npm-package'
 
-  # For CI runs on master, automatically publish packages
-  - bash: |
-        echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > .npmrc
-        cd _build
-        npm publish || true # Ignore publish failures, usually will happen because package already exists
-    displayName: (azure-pipelines-task-lib) npm publish
-    workingDirectory: node
-    condition: and(succeeded(), in(variables['build.reason'], 'IndividualCI', 'BatchedCI', 'Manual'), in(variables['build.sourcebranch'], 'refs/heads/master', 'refs/heads/releases/4.x'))
-    env:
-      NPM_TOKEN: $(npmPublishToken)
+    # For CI runs on master, automatically publish packages
+    - bash: |
```

Comment:
We can use Npm@1 with publish command here.
https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/package/npm?view=azure-devops

Reply:
Good point, but the Npm task doesn't allow use parameters for publishing

Reply:
Yes, it actually uses the service connection, and I guess we already have one for npmjs.org